aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/BaseHandlerStd.java

```diff
@@ -506,4 +510,11 @@ protected boolean shouldSetDefaultVpcSecurityGroupIds(final ResourceModel previo
         }
         return CollectionUtils.isEmpty(desiredState.getVpcSecurityGroupIds());
     }
+
+    private void assertNoDBClusterTerminalStatus(final DBCluster dbCluster) throws CfnNotStabilizedException {
+        final DBClusterStatus status = DBClusterStatus.fromString(dbCluster.status());
+        if (status != null && status.isTerminal()) {
+            throw new CfnNotStabilizedException(new Exception("DBDluster is in state: " + status));
```

Comment:
Can we add `DBClusterIdentifier` to the message so it will be meaningful error for our customers?

Reply:
+1

Reply:
Do messages like these need to go through documentation review?

Reply:
Cluster's physical and logical IDs will be visible through stack events.

Reply:
If customer got error from cloud trail or support got it from only failed call perspective. Rds exceptions mention the cluster or instance id even that customer can get from request parameters that he created.

Reply:
>Rds exceptions mention the cluster or instance id even that customer can get from request parameters that he created.

I do not thinks this holds true. Most, if not all, validation errors on RDS side do not include resource identifier. In addition, including resource identifier in the error message makes the task of grouping the messages together significantly harder.

Reply:
`DBDluster` -> `DBCluster`